setup/config/config_global.go

```diff
@@ -225,3 +234,80 @@ func (c *DNSCacheOptions) Verify(configErrs *ConfigErrors, isMonolith bool) {
 	checkPositive(configErrs, "cache_size", int64(c.CacheSize))
 	checkPositive(configErrs, "cache_lifetime", int64(c.CacheLifetime))
 }
+
+// Consent tracking configuration
+// If either require_at_registration or send_server_notice_to_guest are true, consent
+// messages will be sent to the users.
+type UserConsentOptions struct {
+	// If consent tracking is enabled or not
+	Enabled bool `yaml:"enabled"`
+	// Randomly generated string to be used to calculate the HMAC
+	FormSecret string `yaml:"form_secret"`
+	// Require consent when user registers for the first time
+	RequireAtRegistration bool `yaml:"require_at_registration"`
+	// The name to be shown to the user
+	PolicyName string `yaml:"policy_name"`
+	// The directory to search for *.gohtml templates
+	TemplateDir string `yaml:"template_dir"`
+	// The version of the policy. When loading templates, ".gohtml" template is added as a suffix
+	// e.g: ${template_dir}/1.0.gohtml needs to exist, if this is set to 1.0
+	Version string `yaml:"version"`
+	// Send a consent message to guest users
+	SendServerNoticeToGuest bool `yaml:"send_server_notice_to_guest"`
+	// Default message to send to users
+	ServerNoticeContent struct {
+		MsgType string `yaml:"msg_type"`
+		Body    string `yaml:"body"`
+	} `yaml:"server_notice_content"`
+	// The error message to display if the user hasn't given their consent yet
+	BlockEventsError string `yaml:"block_events_error"`
+	// All loaded templates
+	Templates     *template.Template     `yaml:"-"`
+	TextTemplates *textTemplate.Template `yaml:"-"`
+	// The base URL this homeserver will serve clients on, e.g. https://matrix.org
+	BaseURL string `yaml:"base_url"`
+}
+
+func (c *UserConsentOptions) Defaults() {
+	c.Enabled = false
+	c.RequireAtRegistration = false
+	c.SendServerNoticeToGuest = false
+	c.PolicyName = "Privacy Policy"
+	c.Version = "1.0"
+	c.TemplateDir = "./templates/privacy"
+	c.BaseURL = "http://localhost"
```

Comment:
I wouldn't set this otherwise people could naively set "enabled: true" in the config and then it won't fall over with a missing base URL.